XiEditor/CoreConnection.swift

```diff
@@ -112,9 +112,11 @@ class CoreConnection {
             let currentTime = Date.init()
             dateFormatter.dateFormat = "yyyy-MM-dd-HHMMSS"
             let timeStamp = dateFormatter.string(from: currentTime)
-            
-            let tmpErrLog = self.logDirectory.appendingPathComponent(self.tmpLogName)
-            let timestampedLog = self.logDirectory.appendingPathComponent("XiEditor_\(timeStamp).log")
+
+            guard let tmpErrLog = self.appDelegate.errorLogDirectory?.appendingPathComponent(self.appDelegate.defaultCoreLogName)
```

Comment:
you can write multiple clauses in a single guard statement in swift, separated by commas, as in:

```swift
var maybeBool: Bool?
guard let bool = maybeBool, bool == false else {
    return
}
```